mm2src/adex_cli/src/scenarios/inquire_extentions.rs

```diff
@@ -0,0 +1,64 @@
+use inquire::parser::DEFAULT_BOOL_PARSER;
+use std::str::FromStr;
+
+#[derive(Clone)]
+pub enum InquireOption<T> {
+    Some(T),
+    None,
+}
+
+type OptionalConfirm = InquireOption<bool>;
+
+impl<T> From<InquireOption<T>> for Option<T> {
+    fn from(value: InquireOption<T>) -> Self {
+        match value {
+            InquireOption::None => None,
+            InquireOption::Some(value) => Some(value),
+        }
+    }
+}
+
+impl<T: FromStr> FromStr for InquireOption<T>
+where
+    <T as FromStr>::Err: ToString,
+{
+    type Err = T::Err;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.is_empty() || s.to_lowercase() == "none" {
+            return Ok(InquireOption::None);
+        }
+        T::from_str(s).map(InquireOption::Some)
+    }
+}
+
+impl<T: ToString> ToString for InquireOption<T> {
+    fn to_string(&self) -> String {
+        match self {
+            InquireOption::Some(value) => value.to_string(),
+            InquireOption::None => "None".to_string(),
+        }
+    }
+}
+
+pub type OptionBoolFormatter<'a> = &'a dyn Fn(OptionalConfirm) -> String;
+pub const DEFAULT_OPTION_BOOL_FORMATTER: OptionBoolFormatter = &|ans| -> String {
+    match ans {
+        InquireOption::None => String::new(),
+        InquireOption::Some(true) => String::from("yes"),
+        InquireOption::Some(false) => String::from("no"),
+    }
+};
+
+pub type OptionBoolParser<'a> = &'a dyn Fn(&str) -> Result<InquireOption<bool>, ()>;
+pub const OPTION_BOOL_PARSER: OptionBoolParser = &|ans: &str| -> Result<InquireOption<bool>, ()> {
+    if ans.is_empty() {
+        return Ok(InquireOption::None);
+    }
+    DEFAULT_BOOL_PARSER(ans).map(InquireOption::Some)
+};
+
+pub const DEFAULT_DEFAULT_OPTION_BOOL_FORMATTER: OptionBoolFormatter = &|ans: InquireOption<bool>| match ans {
+    InquireOption::None => String::from("Tap enter to skip/yes/no"),
+    InquireOption::Some(true) => String::from("none/Yes/no"),
```

Comment:
y should be small in `Yes`

Reply:
Actually I shouldn't ). That is an agreement in inquire crate. With capital letter we highlight that is Yes is default.

Reply:
I think the inputs for yes/no/none should be compatible with all possible cases e.g YeS, yes, Y, yEs..

wdyt?

Reply:
The default value for `i_am_seed` is false though, should `DEFAULT_DEFAULT_OPTION_BOOL_FORMATTER` be used with `i_am_seed` in this case?

Reply:
![image](https://user-images.githubusercontent.com/801971/227977976-ea1a74ee-47d9-45fa-ac03-f6e265c68368.png)

Reply:
![image](https://user-images.githubusercontent.com/801971/227978724-61206f03-5406-4049-9e2b-912b6bbb3e01.png)